test/easy-sql-test.js

```diff
@@ -2,13 +2,742 @@
 
 describe('easy-sql-test tests', function() {
+  var emptyDbConfig = {};
 
-  // TODO: write tests
-  it('dummy test', function() {
-    expect(true).to.be.false;
+  describe('constructor test', function() {
+    it('no dbConfig should throw error', function() {
+      var errorMessage = 'easy-sql-test: dbConfig required';
+      try{
+        new EasySqlTest();
+      } catch (err) {
+        expect(err).to.equal(errorMessage)
+      }
+    });
   });
 
+  describe('connectionOpen tests', function() {
+    it('check callback invoked on connection', function() {
+      var connectCallback = sinon.spy(function() {});
+
+      var mssqlStub = {
+        Connection: function() {
+          return {
+            connect: function(callback) {
+              callback();
+            }
+          };
+        }
+      };
+
+      var EasySqlTest = proxyquire('../dist/easy-sql-test.js', {
+        mssql: mssqlStub
+      });
+
+      var easySqlTestProxy = new EasySqlTest(emptyDbConfig);
+
+      easySqlTestProxy.connectionOpen(connectCallback);
+
+      expect(connectCallback.called).to.be.true;
+      expect(connectCallback.callCount).to.equal(1);
+    });
+  });
+
+  describe('connectionClose tests', function(){
+    it('check connection.close() called', function() {
+      var closeFunctionSpy = sinon.spy(function(){});
+      var mssqlStub = {
+        Connection: function() {
+          return {
+            connect: function(){},
+            close: closeFunctionSpy
+          }
+        }
+      };
+
+      var EasySqlTest = proxyquire('../dist/easy-sql-test.js', {
+        mssql: mssqlStub
+      });
+
+      var easySqlTestProxy = new EasySqlTest(emptyDbConfig);
+
+      easySqlTestProxy.connectionOpen();
+      easySqlTestProxy.connectionClose();
+      expect(closeFunctionSpy.called).to.be.true;
+      expect(closeFunctionSpy.callCount).to.equal(1);
+    });
+  });
+
+  describe('property tests', function() {
+
+    it('get connection()', function() {
+      var expectedConnection = {
+        connect: function() {}
+      };
+
+      var mssqlStub = {
+        Connection: function() {
+          return expectedConnection;
+        }
+      };
+
+      var EasySqlTest = proxyquire('../dist/easy-sql-test.js', {
+        mssql: mssqlStub
+      });
+
+      var easySqlTestProxy = new EasySqlTest(emptyDbConfig);
+
+      easySqlTestProxy.connectionOpen();
+      var actualConnection = easySqlTestProxy.connection;
+      expect(actualConnection).to.deep.equal(expectedConnection);
+    });
+
+    it('get dbConfig()', function() {
+
+    	var expected = {
+    		a: 'a',
+    		b: 'b'
+    	};
+
+    	var easySqlTest = new EasySqlTest(expected);
+
+    	var actual = easySqlTest.dbConfig;
+    	expect(expected).to.deep.equal(actual);
+    });
+  });
+
+  describe('_executeStorProc() tests', function(){
+
+    it('no storProcName should throw error', function() {
+  		var errorMessage = 'easy-sql-test: _executeStorProc() requires' +
```

Comment:
I think your editor is messed up. Could you please double check not to use Tabs anywhere in the file please